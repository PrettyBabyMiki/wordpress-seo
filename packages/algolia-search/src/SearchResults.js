/* External dependencies */
import React from "react";
import PropTypes from "prop-types";
import { speak as a11ySpeak } from "@wordpress/a11y";
import styled from "styled-components";
import { __, sprintf } from "@wordpress/i18n";

/* Yoast dependencies */
import { ZebrafiedListTable, Row } from "@yoast/components";
import { colors } from "@yoast/style-guide";

/**
 * The title of the search result item.
 */
const SearchResultTitle = styled.p`
	margin: 0;
	font-size: 1em;
	font-weight: normal;
`;

/**
 * The link to the search result.
 */
const SearchResultLink = styled.a`
	color: ${ colors.$color_black };
	padding: 8px 16px;

	&:hover, &:focus {
		color: ${ colors.$color_pink_dark };
	}
`;

/**
 * Returns a single search result item.
 *
 * @param {Object} props Properties of the SearchResult component.
 * @returns {ReactElement} The rendered search result.
 */
export function SearchResult( props ) {
	const post = props.post;

	return (
		<Row { ...props }>
			<SearchResultLink href={ post.permalink } onClick={ props.onClick }>
				<SearchResultTitle>{ post.post_title }</SearchResultTitle>
			</SearchResultLink>
		</Row>
	);
}

SearchResult.propTypes = {
	onClick: PropTypes.func.isRequired,
	post: PropTypes.object.isRequired,
};

/**
 * Wraps the search results.
 */
const SearchResultsWrapper = styled.div`
	margin-top: 20px;
	clear: both;
`;

/**
 * The no results text paragraph.
 */
const NoResults = styled.p`
	margin-left: 10px;
`;

/**
 * Creates the SearchResults component to render a list of search results.
 *
 * @param {object} props The React props.
 *
 * @returns {ReactElement} The SearchResults component.
 */
class SearchResults extends React.Component {
	/**
	 * Constructs the component and sets its initial state.
	 *
	 * @param {Object} props The props to use for this component.
	 *
	 * @constructor
	 */
	constructor( props ) {
		super( props );

		this.state = {
			results: props.results,
		};
	}

	/**
	 * Handles the situation that there are zero results.
	 *
	 * @returns {ReactElement|null} A message stating that no results were found or null if no search term was provided.
	 */
	handleZeroResults() {
		if ( this.props.searchString !== "" ) {
			return this.renderNoResultsFound();
		}

		return null;
	}

	/**
	 * Renders a no results found text.
	 *
	 * @returns {ReactElement} The no results found text.
	 */
	renderNoResultsFound() {
		const noResultsText = __( "No results found.", "yoast-components" );

		a11ySpeak( noResultsText );

		return (
			<NoResults>
				{ noResultsText }
			</NoResults>
		);
	}

	/**
	 * Maps the results to SearchResult components.
	 *
	 * @param {Object} results The results returned by Algolia.
	 *
	 * @returns {Array} Array containing the mapped search results.
	 */
	resultsToSearchItem( results ) {
		return results.map( ( result, index ) => {
			return <SearchResult
				rowHeight="32px"
				key={ result.objectID }
				post={ result }
				// Note: this passes the onClick but actually also attaches a click event on the LI element.
				onClick={ ( event ) => {
					event.preventDefault();
					event.stopPropagation();
					this.props.onClick( index );
				} }
			/>;
		} );
	}

	/**
	 * Renders the search results list.
	 *
	 * @returns {ReactElement} Either returns the search results, or a message that no results were found.
	 */
	render() {
		const resultsCount = this.props.results.length;

		// Check whether no results are returned.
		if ( resultsCount <= 0 ) {
			return this.handleZeroResults();
		}

		const foundResultsText = sprintf( __( "Number of results found: %d", "yoast-components" ), resultsCount );

		a11ySpeak( foundResultsText );

		return (
			<SearchResultsWrapper>
				<h2 className="screen-reader-text">
					{ __( "Search results", "yoast-components" ) }
				</h2>
				<ZebrafiedListTable>
					{ this.resultsToSearchItem( this.props.results ) }
				</ZebrafiedListTable>
			</SearchResultsWrapper>
		);
	}
}

SearchResults.propTypes = {
	onClick: PropTypes.func.isRequired,
	searchString: PropTypes.string,
	results: PropTypes.array,
};

SearchResults.defaultProps = {
	searchString: "",
	results: [],
};

export default SearchResults;
