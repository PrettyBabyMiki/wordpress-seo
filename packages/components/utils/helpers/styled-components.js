/**
 * Returns a function that returns the correct style based on if the current
 * theme is right to left.
 *
 * This is determined by the `isRtl` property in the styled components theme.
 *
 * @param {string} left  Style to return if the theme is left to right.
 * @param {string} right Style to return if the theme is right to left.
 *
 * @returns {Function} A function that returns the right styled based on the
 * 					   theme in the props.
 */
export function getRtlStyle( left, right ) {
	return ( props ) => {
		return ( props.theme.isRtl ? right : left );
	};
}
