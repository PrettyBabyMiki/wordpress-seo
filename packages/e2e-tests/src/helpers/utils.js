/**
 * WordPress e2e utilities
 */
import {
    installPlugin,
    activatePlugin,
    visitAdminPage,
    pressKeyWithModifier,
} from "@wordpress/e2e-test-utils";

import { addQueryArgs } from '@wordpress/url';

export const createCustomPostType = async( slug ) => {
   
}