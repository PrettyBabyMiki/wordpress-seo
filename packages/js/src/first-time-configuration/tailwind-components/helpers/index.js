import classNames from "classnames";

export const socialMedia = [
	{ name: "Facebook", placeholder: "E.g. https://www.facebook.com/yoast" },
	{ name: "Instagram", placeholder: "E.g. https://www.instagram.com/yoast" },
	{ name: "LinkedIn", placeholder: "E.g. https://www.linkedin.com/yoast" },
	{ name: "MySpace", placeholder: "E.g. https://www.myspace.com/yoast" },
	{ name: "Pinterest", placeholder: "E.g. https://www.pinterest.com/yoast" },
	{ name: "SoundCloud", placeholder: "E.g. https://www.soundcloud.com/yoast" },
	{ name: "Tumblr", placeholder: "E.g. https://www.tumblr.com/yoast" },
	{ name: "Twitter", placeholder: "E.g. https://www.twitter.com/yoast" },
	{ name: "YouTube", placeholder: "E.g. https://www.youtube.com/yoast" },
	{ name: "Wikipedia", placeholder: "E.g. https://www.wikipedia.com/yoast" },
];

/**
 * Creates the error ID for the error component.
 *
 * @param {string} inputId The id of the input component.
 *
 * @returns {string} The ID for the error component.
 */
export const getErrorId = ( inputId ) => `error-${ inputId }`;

/**
  * Get props needed to properly display an error in input components.
  *
  * @param {string} inputId The id of the input component.
  * @param {ValidationError} error The error object.
  * @param {boolean} error.isVisible The error object.
  *
  * @returns {Object} Object containing relevant props for displaying.
  */
export const getErrorAriaProps = ( inputId, { isVisible } ) => isVisible ? {
	"aria-invalid": true,
	"aria-describedby": getErrorId( inputId ),
} : {};

/**
 * Helper function to get active styles for select options.
 *
 * @param {boolean} options.active Whether the option is active.
 *
 * @returns {string} Styles for an active option.
 */
export function getOptionActiveStyles( { active, selected } ) {
	return classNames(
		"yst-relative yst-cursor-default yst-select-none yst-py-2 yst-pl-3 yst-pr-9 yst-my-0",
		selected && "yst-bg-primary-500 yst-text-white",
		( active && ! selected ) && "yst-bg-primary-200 yst-text-gray-700",
		( ! active && ! selected ) && "yst-text-gray-700"
	);
}
