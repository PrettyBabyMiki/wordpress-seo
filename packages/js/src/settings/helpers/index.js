export * from "./i18n";
export * from "./user-social-profiles";
export * from "./search";
export * from "./submit";
export * from "./validation";
