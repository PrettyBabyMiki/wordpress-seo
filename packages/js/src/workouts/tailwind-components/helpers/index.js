import classNames from "classnames";

/**
 * Creates the error ID for the error component.
 *
 * @param {string} inputId The id of the input component.
 *
 * @returns {string} The ID for the error component.
 */
export const getErrorId = ( inputId ) => `error-${ inputId }`;

/**
  * Get props needed to properly display an error in input components.
  *
  * @param {string} inputId The id of the input component.
  * @param {ValidationError} error The error object.
  * @param {boolean} error.isVisible The error object.
  *
  * @returns {Object} Object containing relevant props for displaying.
  */
export const getErrorAriaProps = ( inputId, { isVisible } ) => isVisible ? {
	"aria-invalid": true,
	"aria-describedby": getErrorId( inputId ),
} : {};

/**
 * Helper function to get active styles for select options.
 *
 * @param {boolean} options.active Whether the option is active.
 *
 * @returns {string} Styles for an active option.
 */
export function getOptionActiveStyles( { active, selected } ) {
	return classNames(
		"yst-relative yst-cursor-default yst-select-none yst-py-2 yst-pl-3 yst-pr-9 yst-my-0",
		selected && "yst-bg-primary-500 yst-text-white",
		( active && ! selected ) && "yst-bg-primary-200 yst-text-gray-900",
		( ! active && ! selected ) && "yst-text-gray-900"
	);
}
