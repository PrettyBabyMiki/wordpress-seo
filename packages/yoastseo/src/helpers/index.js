import scoreToRating from "../interpreters/scoreToRating";
import { measureTextWidth } from "./createMeasurementElement";
import getLanguagesWithWordFormSupport from "./getLanguagesWithWordFormSupport";

export {
	measureTextWidth,
	scoreToRating,
	getLanguagesWithWordFormSupport,
};

export * from "./inRange";
