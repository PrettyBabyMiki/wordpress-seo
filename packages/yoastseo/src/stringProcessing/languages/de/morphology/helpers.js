import { flatten } from "lodash-es";

/**
 * Returns combined list of all German verb prefixes, sorted by length (descending).
 *
 * @param {Object} verbPrefixDataGerman The prefix data for German verbs.
 *
 * @returns {Array<string>} All German verb prefixes, sorted by length (descending).
 */
export function allGermanVerbPrefixesSorted( verbPrefixDataGerman ) {
	const allPrefixes = flatten( Object.values( verbPrefixDataGerman ) );

	return allPrefixes.sort( ( a, b ) => b.length - a.length || a.localeCompare( b ) );
}
