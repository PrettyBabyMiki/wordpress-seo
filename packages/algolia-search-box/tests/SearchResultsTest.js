import React from "react";

import { createComponentWithIntl } from "@yoast/helpers";
import SearchResults from "../src/SearchResults";

const post = { permalink: "https://kb.yoast.com/kb/passive-voice/", postTitle: "Post Title", objectID: 1 };

test( "the SearchResults component with results matches the snapshot", () => {
	const component = createComponentWithIntl(
		<SearchResults
			onClick={ () => {} }
			post={ post }
			showDetail={ () => {} }
			searchString="Test"
			results={ [ post ] }
			noResultsText="No Results"
			foundResultsText="These are your %d results"
		/>
	);

	const tree = component.toJSON();
	expect( tree ).toMatchSnapshot();
} );

test( "the SearchResults component without results matches the snapshot", () => {
	const component = createComponentWithIntl(
		<SearchResults
			onClick={ () => {} }
			post={ {} }
			showDetail={ () => {} }
			searchString="Test"
			results={ [] }
			noResultsText="No Results"
			foundResultsText="These are your %d results"
		/>
	);

	const tree = component.toJSON();
	expect( tree ).toMatchSnapshot();
} );
