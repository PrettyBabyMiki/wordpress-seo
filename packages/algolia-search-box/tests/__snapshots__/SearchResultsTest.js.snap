// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`the SearchResults component with results matches the snapshot 1`] = `
<div
  className="SearchResults__SearchResultsWrapper-sc-1s841j3-2 kLUjtt"
>
  <h2
    className="screen-reader-text"
  >
    Search results
  </h2>
  <ul
    className="ListTable__List-sc-1oyt4xr-0 gbioMK"
    role="list"
  >
    <li
      className="Row-ehfonz-0 jvCcjA"
      onClick={[Function]}
    >
      <a
        className="SearchResults__SearchResultLink-sc-1s841j3-1 bmXCvI"
        href="https://kb.yoast.com/kb/passive-voice/"
        onClick={[Function]}
      >
        <p
          className="SearchResults__SearchResultTitle-sc-1s841j3-0 kcvibD"
        />
      </a>
    </li>
  </ul>
</div>
`;

exports[`the SearchResults component without results matches the snapshot 1`] = `
<p
  className="SearchResults__NoResults-sc-1s841j3-3 gxkCVR"
>
  No results found.
</p>
`;
